src/shogun/lib/any.h

```diff
@@ -212,6 +283,105 @@ namespace shogun
             policy->clear(&storage);
         }
 
+		template <class Archive, class Type>
+		void cereal_save_helper(Archive& ar) const
+		{
+			ar(*(reinterpret_cast<Type*>(storage)));
+		}
+
+		template<class Archive>
+		void cereal_save(Archive& ar) const
+		{
+			ar (m_datatype.e_containertype);
+			ar (m_datatype.e_primitivetype);
+			switch (m_datatype.e_containertype) {
+			case serial::EnumContainerType::CT_PRIMITIVE:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_save_helper<Archive, int32_t>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
+					cereal_save_helper<Archive, float64_t>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_UNDEFINED:
+					SG_SERROR("Type error: undefined data type cannot be serialized.\n");
+					break;
+				}
+				break;
+			case serial::EnumContainerType::CT_SGVECTOR:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_save_helper<Archive, SGVector<int32_t> >(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
+					cereal_save_helper<Archive, SGVector<float64_t> >(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_UNDEFINED:
+					SG_SERROR("Type error: undefined data type cannot be serialized.\n");
+					break;
+				}
+				break;
+			case serial::EnumContainerType::CT_SGMATRIX:
+				SG_SWARNING("SGMatrix serializatino method not implemented.\n");
+				break;
+			case serial::EnumContainerType::CT_UNDEFINED:
+				SG_SERROR("Type error: undefined container type cannot be serialized.\n");
+				break;
+			}
+		}
+
+		template <class Archive, class Type>
+		void cereal_load_helper(Archive& ar)
+		{
+			Type temp;
+			ar(temp);
+			policy->clear(&storage);
+			policy->set(&storage, &temp);
+		}
+
+		template<class Archive>
+		void cereal_load(Archive& ar)
+		{
+			ar(m_datatype.e_containertype);
+			ar(m_datatype.e_primitivetype);
+			switch (m_datatype.e_containertype) {
+			case serial::EnumContainerType::CT_PRIMITIVE:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_load_helper<Archive, int32_t>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
+					cereal_load_helper<Archive, float64_t>(ar);
+					break;
+				default:
+					SG_SERROR("Error: undefined data type cannot be loaded.\n");
+					break;
+				}
+				break;
+
+			case serial::EnumContainerType::CT_SGVECTOR:
+				switch (m_datatype.e_primitivetype) {
+				case serial::EnumPrimitiveType::PT_INT_32:
+					cereal_load_helper<Archive, SGVector<int32_t>>(ar);
+					break;
+				case serial::EnumPrimitiveType::PT_FLOAT_64:
```

Comment:
Can we also at some point have a benchmark of how fast it is to store classes with the new vs the old framework?